lib/wda/webdriveragent.js

```diff
@@ -221,8 +223,12 @@ class WebDriverAgent {
     await resetXCTestProcesses(this.device.udid, !this.realDevice, {wdaLocalPort: this.url.port});
 
     if (this.realDevice) {
-      this.iproxy = new iProxy(this.device.udid, this.url.port, WDA_AGENT_PORT);
-      await this.iproxy.start();
+      if ([WDA_BASE_URL, WDA_BASE_URL_IP].includes(this.wdaBaseUrl)) {
+        this.iproxy = new iProxy(this.device.udid, this.url.port, WDA_AGENT_PORT);
```

Comment:
do we need to check if this.iproxy is set in the teardown?

Reply:
do you mean https://github.com/appium/appium-xcuitest-driver/pull/941/files#diff-dd1c1e9a4d2cba48fc21e5d2c50abcafR274 in `quite()`?
```
    if (this.iproxy) {
      await this.iproxy.quit();
    }
```

Reply:
yep, exactly